seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/physical/PhysicalPlan.java

```diff
@@ -17,38 +17,134 @@
 
 package org.apache.seatunnel.engine.server.dag.physical;
 
-import org.apache.seatunnel.engine.server.task.TaskGroupInfo;
+import org.apache.seatunnel.engine.common.utils.NonCompletableFuture;
```

Comment:
I have a little doubt, why put the scheduling-related logic directly in the physical plan, why not create a new class to do this?

Reply:
> I have a little doubt, why put the scheduling-related logic directly in the physical plan, why not create a new class to do this?

I refer to Flink, spark and other computing engines, which manage the job state in PyhsicalPlan. I understand that this is reasonable. The physical execution plan is not only used to represent a DAG relationship. The execution plan should be responsible for job execution and state manager.

Reply:
Got it